LoRaEngine/LoraKeysManagerFacade/IoTCentralImp/DeviceProvisioningHelper.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoraKeysManagerFacade.IoTCentralImp
+{
+    using System;
+    using System.Security.Cryptography;
+    using System.Text;
+    using LoraKeysManagerFacade.IoTCentralImp.Definitions;
+    using Microsoft.Azure.Devices.Provisioning.Client;
+    using Microsoft.Azure.Devices.Provisioning.Client.Transport;
+    using Microsoft.Azure.Devices.Shared;
+
+    public class DeviceProvisioningHelper : IDeviceProvisioningHelper
+    {
+        private readonly string deviceProvisioningEndpoint;
+        private readonly string provisioningScopeId;
+        private readonly string deviceProvisioningPrimaryKey;
+        private readonly string deviceProvisioningSecondaryKey;
+
+        public DeviceProvisioningHelper(string provisioningScopeId, string primaryKey, string secondaryKey, string deviceProvisioningEndpoint = "global.azure-devices-provisioning.net")
+        {
+            this.deviceProvisioningEndpoint = deviceProvisioningEndpoint;
+            this.provisioningScopeId = provisioningScopeId;
+            this.deviceProvisioningPrimaryKey = primaryKey;
+            this.deviceProvisioningSecondaryKey = secondaryKey;
+        }
+
+        public SymmetricKeyAttestation ProvisionDevice(string deviceId, out string assignedIoTHubHostname)
+        {
+            var attestation = this.ComputeAttestation(deviceId);
+
+            using var transportHandler = new ProvisioningTransportHandlerHttp();
+            using var securityProvider = new SecurityProviderSymmetricKey(deviceId, primaryKey: attestation.SymmetricKey.PrimaryKey, secondaryKey: attestation.SymmetricKey.SecondaryKey);
+            var provisioningClient = ProvisioningDeviceClient.Create(
+              this.deviceProvisioningEndpoint,
+              this.provisioningScopeId,
+              securityProvider,
+              transportHandler);
+
+            var registerResult = provisioningClient.RegisterAsync().Result;
```

Comment:
We should use async here and rename the method as Async

Reply:
This is not as simple as that, this method returns the assigned iot hub as an output parameter.
This is not compatible with async method...

Reply:
What about creating the IoT CentralDevice object in there instead of returning a null value? I am not a big fan of blocking the thread here

Reply:
I implemented your suggestion.